### PR TITLE
[commhistory-daemon] Reflect ofono voicemail waiting in notification. Contributes to MER#1056

### PR DESCRIPTION
**Pull request by matthewvogt**
Create a notification that exposes the value of the ofono 'voicemail message waiting' state.

### CODE REVIEW

File: src/notificationmanager.cpp [hunk 1]
```diff
@@ -771,3 +768,69 @@ void NotificationManager::slotNgfEventFinished(quint32 id)
         m_ngfEvent = 0;
 }
 
+void NotificationManager::slotVoicemailWaitingChanged(bool waiting)
+{
+    const QString category(QStringLiteral("x-nemo.messaging.voicemail-waiting"));
```
Comment (thp) [inline src/notificationmanager.cpp]: Do we have a named constant for `"x-nemo.messaging.voicemail-waiting"` in some headers? If not, should we have it and use it here?

Comment (matthewvogt) [inline src/notificationmanager.cpp]: No, I just added it.  I'll create one.

File: src/notificationmanager.cpp [hunk 2]
```diff
@@ -771,3 +768,67 @@ void NotificationManager::slotNgfEventFinished(quint32 id)
...
+            if (waiting) {
+                // The notification is already present; do nothing
+                currentId = n->replacesId();
+            } else {
+                // Close this notification
+                n->close();
+            }
+        }
+    }
+    qDeleteAll(notifications);
```
Comment (thp) [inline src/notificationmanager.cpp]: Maybe add `notifications.clear();` below the `qDeleteAll()` call to make sure nobody accidentally touches any dangling pointers in the future.

**Comment by thp** [comment]
Rest LGTM.

File: src/notificationmanager.cpp (hunk 2 above)
Comment (matthewvogt) [inline src/notificationmanager.cpp]: Thanks, fixed

**Comment by thp** [comment]
LGTM.

**Pull request merged**